File: web/src/app/types/restaurant.ts
export interface Restaurant {
  id: string;
  name: string;
  cuisine: string;
  priceRange: "$" | "$$" | "$$$" | "$$$$";
  rating: number;
  reviewCount: number;
  address: string;
  distance: number; // in miles
  image: string;
  hours: string;
  phone: string;
  description: string;
  latitude: number;
  longitude: number;
  tags: string[];
}

export interface Review {
  id: string;
  restaurantId: string;
  userName: string;
  userAvatar: string;
  rating: number;
  date: string;
  comment: string;
  images?: string[];
}

export interface User {
  id: string;
  name: string;
  email: string;
  avatar: string;
  favoriteRestaurants: string[];
}

File: backend/restoradar/src/main/java/edu/cit/chan/restoradar/entity/Restaurant.java
package edu.cit.chan.restoradar.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "restaurants")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Restaurant {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    // "P", "PP", "PPP"
    @Column(name = "price_range", nullable = false)
    private String priceRange;

    private String cuisine;

    private String address;

    @Column(nullable = false)
    private Double latitude;

    @Column(nullable = false)
    private Double longitude;

    @Column(name = "contact_number")
    private String contactNumber;

    private String website;

    // Stored as JSON string: {"monday":"11:00-22:00", ...}
    @Column(columnDefinition = "TEXT")
    private String hours;

    @Column(nullable = false)
    private Double rating = 0.0;

    @Column(name = "review_count", nullable = false)
    private Integer reviewCount = 0;

    // Stored as comma-separated URLs (or use @ElementCollection for a join table)
    @ElementCollection
    @CollectionTable(name = "restaurant_photos", joinColumns = @JoinColumn(name = "restaurant_id"))
    @Column(name = "photo_url")
    private List<String> photos;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id")
    private UserEntity owner;

    @Column(nullable = false)
    private Boolean approved = false;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private Instant createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private Instant updatedAt;
}